### PR TITLE
fix: claiming firefly red packet and the cover

### PR DESCRIPTION
**Pull request by UncleBill**
MF-0000

### CODE REVIEW

File: packages/plugins/RedPacket/src/SiteAdaptor/hooks/useAvailabilityComputed.ts [hunk 1]
```diff
@@ -22,16 +25,24 @@ export function useAvailabilityComputed(account: string, payload: RedPacketJSONP
         chainId: parsedChainId,
     })
 
-    const { data: password = payload.password } = useQuery({
+    const author = usePostInfoDetails.author()
+    const platform = usePlatformType()
+    const signedMsg = useSignedMessage({
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/hooks/useAvailabilityComputed.ts]: Handle in the following useQuery once.

File: packages/plugins/RedPacket/src/SiteAdaptor/hooks/useAvailabilityComputed.ts [hunk 2]
```diff
@@ -22,16 +25,24 @@ export function useAvailabilityComputed(account: string, payload: RedPacketJSONP
...
+        account,
+        contractVersion: payload.contract_version,
+        password: payload.password,
+        rpid: payload.rpid,
+        profile: platform ? { platform, profileId: author?.userId || '' } : undefined,
+    })
+    const { data: password = signedMsg } = useQuery({
```
Comment (guanbinrui) [inline packages/plugins/RedPacket/src/SiteAdaptor/hooks/useAvailabilityComputed.ts]: This looks strange to me. A password and a signature are not the same thing.

Comment (UncleBill) [reply packages/plugins/RedPacket/src/SiteAdaptor/hooks/useAvailabilityComputed.ts]: Actually, it's the `signedMsg` parameter to the contract.

File: packages/plugins/RedPacket/src/SiteAdaptor/hooks/useAvailabilityComputed.ts (hunk 1 above)
Comment (UncleBill) [reply packages/plugins/RedPacket/src/SiteAdaptor/hooks/useAvailabilityComputed.ts]: Would like to reused this.

File: packages/web3-providers/src/helpers/fetchJSON.ts [hunk 3]
```diff
@@ -8,7 +8,17 @@ export async function fetchJSON<T = unknown>(
     init?: RequestInit,
     options?: NextFetchersOptions,
 ): Promise<T> {
-    const response = await fetch(input, init, getNextFetchers(options))
+    const response = await fetch(
+        input,
+        {
+            headers: {
+                'Content-Type': 'application/json',
```
Comment (guanbinrui) [inline packages/web3-providers/src/helpers/fetchJSON.ts]: It's not safe to introduce this change.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**